Tournament.cpp

```diff
@@ -0,0 +1,221 @@
+				{
+					if (team->m_place == 1)
+						firstTeams.push(team);
+					if (team->m_place == 2)
+						secondTeams.push(team);
+					if(team->m_place == 3)
+						preparationThirdTeams.push_back(team);
+				});
+		}
+		//Find four best teams in third places
+		sort(preparationThirdTeams.begin(), preparationThirdTeams.end(), [](std::shared_ptr<Team>& lhs, std::shared_ptr<Team>& rhs)
+			{
+				return std::tie(lhs->m_points, lhs->m_goal_difference) >
+					std::tie(rhs->m_points, rhs->m_goal_difference);
+			});
+		//Remove fifth and sixth teams
+		for (int i = 0; i < 2; ++i)
+		{
+			preparationThirdTeams.pop_back();
+			preparationThirdTeams.shrink_to_fit();
+		}
+		//put commands in the queue
+		for (auto it : preparationThirdTeams)
+		{
+			thirdTeams.push(it);
+		}
+
+		//Create a playoff grid
+		for (int i = 0; i < 2; ++i)
+		{
+			//A2-B2 and D2-F2
+			std::shared_ptr<Team> firstTmpTeam = secondTeams.front();
+			secondTeams.pop();
+			std::shared_ptr<Team> secondTmpTeam = secondTeams.front();
+			secondTeams.pop();
+			if (i == 0)
+				left_side.push_back(SetMatch(firstTmpTeam, secondTmpTeam));
+			if (i == 1)
+				right_side.push_back(SetMatch(firstTmpTeam, secondTmpTeam));
+			//A1-C2 and D1-F2
+			firstTmpTeam = firstTeams.front();
+			firstTeams.pop();
+			secondTmpTeam = secondTeams.front();
+			secondTeams.pop();
+			if (i == 0)
+				right_side.push_back(SetMatch(firstTmpTeam, secondTmpTeam));
+			if (i == 1)
+				left_side.push_back(SetMatch(firstTmpTeam, secondTmpTeam));
+			//B1-third-2 and E1-third-4
+			std::shared_ptr<Team> thirdTmpTeam = thirdTeams.front(); //get third-1 and third-4 from the queue
+			thirdTeams.pop();
+			firstTmpTeam = firstTeams.front();
+			firstTeams.pop();
+			secondTmpTeam = thirdTeams.front();
+			thirdTeams.pop();
+			if (i == 0)
+				right_side.push_back(SetMatch(firstTmpTeam, secondTmpTeam));
+			if (i == 1)
+				left_side.push_back(SetMatch(firstTmpTeam, secondTmpTeam));
+			//C1-third-1 and F1-third-3
+			firstTmpTeam = firstTeams.front();
+			firstTeams.pop();
+			if (i == 0)
+				left_side.push_back(SetMatch(firstTmpTeam, thirdTmpTeam));
+			if (i == 1)
+				right_side.push_back(SetMatch(firstTmpTeam, thirdTmpTeam));
+		}
+		int count = 0;
+		for (auto it : left_side)
+		{
+			std::cout << "Match #" << ++count << std::endl;
+			std::cout << it.first_team << std::endl << it.second_team << std::endl << std::endl;
+		}
+		for (auto it : right_side)
+		{
+			std::cout << "Match #" << ++count << std::endl;
+			std::cout << it.first_team << std::endl << it.second_team << std::endl << std::endl;
+		}
+	}
+	void Tournament::PrintParticipants()
+	{
+		for (auto it : m_participants)
+			std::cout <<std::setw(10)<< it->m_name << "\t" << it->m_fifa_rang << "\t" << it->m_fifa_points << std::endl;
+	}
+
+	void Tournament::PrintGroup()
+	{
+		for (auto it : m_groups)
+			std::cout << it << std::endl;
+	}
+
+	Match Tournament::SetMatch(std::shared_ptr<Team> lhs, std::shared_ptr<Team> rhs)
```

Comment:
Isn't CreateMatch instead of SetMatch? And it seems like it can be const. And why is it a non-static method of Tournament when it does not actually use **this**?